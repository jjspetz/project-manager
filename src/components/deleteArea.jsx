import React, { Component } from 'react';
// Material UI imports
import RaisedButton from 'material-ui/RaisedButton';
import Delete from 'material-ui/svg-icons/action/delete';
// react dnd
import { ItemTypes } from '../constants/constants';
import { DropTarget } from 'react-dnd';
// firebase
import database, {User} from '../fire.js'

const deleteTask = {
  drop(props, monitor, connect) {
    console.log(monitor.getItem().id);
    database.ref('users/' + User.user.uid).child(monitor.getItem().id).remove();
  }
};

function collect(connect, monitor) {
  return {
    connectDropTarget: connect.dropTarget(),
    isOver: monitor.isOver()
  };
}

class DeleteArea extends Component {
  render() {
    const { connectDropTarget} = this.props;
    return connectDropTarget(
      <div>
        <RaisedButton
          className='delete'
          icon={<Delete/>}
          primary={true}
          disabled={true}
        />
      </div>
    );
  }
}

export default DropTarget(ItemTypes.TASK, deleteTask, collect)(DeleteArea);
