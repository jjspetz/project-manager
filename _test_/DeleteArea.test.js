import React from 'react';
import { shallow } from 'enzyme'
import { DeleteArea } from '../src/components/deleteArea.jsx';
import toJson from 'enzyme-to-json';

const faux = el => el;
const wrapper = shallow(<DeleteArea
                              connectDropTarget={faux} />);

describe('DeleteArea', () => {
  it('should render', () => {
    expect(wrapper).toBeTruthy();
  });
  it('should contain a div', () => {
    expect(wrapper.find('div')).toBeTruthy();
  });
  it('should match snapshot', () => {
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
  });
});
