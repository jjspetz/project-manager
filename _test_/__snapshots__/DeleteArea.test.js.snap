// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteArea should match snapshot 1`] = `
<div>
  <RaisedButton
    className="delete"
    disabled={true}
    fullWidth={false}
    icon={<ActionDelete />}
    labelPosition="after"
    primary={true}
    secondary={false}
  />
</div>
`;
